src/main/java/Grapie/Commands/TaskList.java

```diff
@@ -0,0 +1,223 @@
+     */
+    public TaskList(List<Task> load, Storage storage) {
+        //contains the task list e.g., it has operations to add/delete tasks in the list
+
+        storingList = load;
+        this.storage = storage;
+    }
+
+    /**
+     * Check for todo, event, deadline and add to list correspondingly.
+     * Also checks for invalid input.
+     *
+     * @param inputStr the user input.
+     * @throws GrapieExceptions Errors thrown.
+     * @throws IOException      Throws away the exception.
+     */
+    public String addToList(String inputStr) throws GrapieExceptions, IOException {
+        if (inputStr.contains("todo")) {
+            if (inputStr.substring(0, 4).equals("todo") && inputStr.length() > 5) {
+                String detailsStr = inputStr.substring(5, inputStr.length());
+                String checkIfTodoIsEmpty = detailsStr.replaceAll("\\s", "");
+
+                assert(checkIfTodoIsEmpty.length() >= 0);
+
+                if (checkIfTodoIsEmpty.length() == 0) {
+                    //That means it is empty behing todo
+                    throw new GrapieExceptions(ErrorMsg.emptyDescriptionError);
+                } else {
+                    Task todo = new Todo(detailsStr);
+                    storingList.add(todo);
+                    String result = Ui.printAddingTask(todo, storingList.size());
+
+                    //store into hard disk
+                    storage.convertAndStore(todo, "T", "");
+
+                    return result;
+                }
+            } else {
+                throw new GrapieExceptions(ErrorMsg.emptyDescriptionError);
+            }
+
+        } else if (inputStr.contains("event")) {
+            if (inputStr.substring(0, 5).equals("event") && inputStr.length() > 6) {
+                inputStr = inputStr.trim();
+                String[] eventAndTime = inputStr.substring(6, inputStr.length()).split(" /at ");
+
+                assert(eventAndTime.length >= 0);
+
+                if (eventAndTime.length <= 1) {
+                    //not able to split string properly
+                    throw new GrapieExceptions(ErrorMsg.eventFormatError);
+                } else {
+                    Event event = new Event(eventAndTime[0], eventAndTime[1]);
+                    storingList.add(event);
+
+                    //printing
+                    String result = Ui.printAddingTask(event, storingList.size());
+
+                    //store into hard disk
+                    storage.convertAndStore(event, "E", eventAndTime[1]);
+                    return result;
+                }
+            } else {
+                throw new GrapieExceptions(ErrorMsg.emptyDescriptionError);
+            }
+
+        } else if (inputStr.contains("deadline")) {
+            if (inputStr.substring(0, 8).equals("deadline") && inputStr.length() > 9) {
+
+                inputStr.trim();
+                String[] eventAndTime = inputStr.substring(9, inputStr.length()).split(" /by ");
```

Comment:
- For instance, this is a low-level string operation.
- Additionally, some operations can be extracted to a method and just invoked here.